mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -90,10 +92,22 @@ class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
   @Since("2.0.0")
   def setBinary(value: Boolean): this.type = set(binary, value)
 
+  /**
+   * Returns the index of the input term.
+   */
+  @Since("2.3.0")
+  def indexOf(term: Any): Int = {
+    if (hashingTF != null) {
+      hashingTF.indexOf(term)
```

Comment:
You can move the `feature.HashingTF` initialization into constructor.
Otherwise the `indexOf` will throw error when `transform` not called, it is weird to user.